### PR TITLE
Support runtime configuration of all current components

### PR DESCRIPTION
**Pull request by sethvargo**
This resolves most TODOs in setup by resolving runtime configuration values for:

- Key Management
- Secret Management
- Storage (already done, but refactored a bit)
- Observability (already done, but tuned a bit)

And documents it 😄.

### CODE REVIEW

File: internal/admin/config.go [hunk 1]
```diff
@@ -23,24 +23,45 @@ import (
...
 
-var _ setup.DBConfigProvider = (*Config)(nil)
+var _ setup.DatabaseConfigProvider = (*Config)(nil)
+var _ setup.KeyManagerConfigProvider = (*Config)(nil)
+var _ setup.SecretManagerConfigProvider = (*Config)(nil)
+var _ setup.BlobstoreConfigProvider = (*Config)(nil)
 
 type Config struct {
+	Database      *database.Config
+	KeyManager    *signing.Config
```
Comment (mikehelmick) [inline internal/admin/config.go:39]: is the admin console using these dependencies now?

Comment (mikehelmick) [reply internal/admin/config.go:39]: well, I guess implicitly secret manager was being used before.

Comment (sethvargo) [reply internal/admin/config.go:39]: Yea I put Secret Manager everywhere we have a database. I can remove KeyManager, but it doesn't hurt...

Comment (mikehelmick) [reply internal/admin/config.go:39]: agreed, doesn't hurt. may be useful in the future even :)

**Pull request merged**